src/Altinn.Common/Altinn.EFormidlingClient/Altinn.EFormidlingClient.Tests/UnitTest/EFormidlingClientUnitTest.cs

```diff
@@ -0,0 +1,202 @@
+using System;
+using System.Collections.Generic;
+using System.Configuration;
+using System.IO;
+using System.Net.Http;
+using System.Text;
+using System.Text.Json;
+using System.Xml.Serialization;
+using Altinn.Common.EFormidlingClient;
+using Altinn.Common.EFormidlingClient.Configuration;
+using Altinn.Common.EFormidlingClient.Models;
+using Altinn.Common.EFormidlingClient.Models.SBD;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Console;
+using Microsoft.Extensions.Logging.Debug;
+using Xunit;
+using Arkivmelding = Altinn.Common.EFormidlingClient.Models.Arkivmelding;
+
+namespace Altinn.EFormidlingClient.Tests.ClientUnitTest
+{
+    /// <summary>
+    /// Represents a collection of unit test, testing the<see cref="EFormidlingClientUnitTest"/> class.
+    /// </summary>
+    public class EFormidlingClientUnitTest : IClassFixture<FixtureUnit>
+    {
+        private readonly ServiceProvider _serviceProvider;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="EFormidlingClientUnitTest"/> class.
+        /// </summary>
+        /// <param name="fixture">Fixture for setup</param>
+        public EFormidlingClientUnitTest(FixtureUnit fixture)
+        {
+            _serviceProvider = fixture.ServiceProvider;
+        }
+
+        /// <summary>
+        /// Test valid sbd json from file
+        /// </summary>
+        [Fact]
+        public void Is_Valid_Json()
+        {
+            var jsonString = File.ReadAllText(@"TestData\sbd.json");
+            var json = JsonSerializer.Deserialize<StandardBusinessDocument>(jsonString);
+
+            Assert.NotNull(json);
+        }
+
+        /// <summary>
+        /// Test invalid sbd json from file
+        /// </summary>
+        [Fact]
+        public void Is_Not_Valid_Json()
+        {
+            var jsonString = File.ReadAllText(@"TestData\sbdInvalid.json");
+            var json = JsonSerializer.Deserialize<StandardBusinessDocument>(jsonString);
+
+            Assert.NotNull(json);
+        }
+
+        /// <summary>
+        /// Test valid xml arkivmelding from file
```

Comment:
I'd love a more explanatory comment here.. We are verifying that a file in with the expected content is successfully parsed into the model that we have defined in the project.

Reply:
Would also love a test the other way.. From a C# object, to through the xml serializer with some sort of validation of the final result